gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspConfigurations.kt

```diff
@@ -0,0 +1,129 @@
+package com.google.devtools.ksp.gradle
+
+import org.gradle.api.InvalidUserCodeException
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.jetbrains.kotlin.gradle.dsl.*
+import org.jetbrains.kotlin.gradle.plugin.*
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinJvmAndroidCompilation
+
+/**
+ * Creates and retrieves ksp-related configurations.
+ */
+class KspConfigurations(private val project: Project) {
+    companion object {
+        private const val PREFIX = "ksp"
+    }
+
+    // Store all ksp configurations for quick retrieval.
+    private val kotlinConfigurations = mutableMapOf<KotlinSourceSet, Configuration>()
+    private val androidConfigurations = mutableMapOf<String, Configuration>()
```

Comment:
I'm not sure how to do this without complicating the logic. I'm using string for Android to keep all AGP types within `AndroidPluginIntegration` class (see docs of the class, AGP is `compileOnly`).

If we use the set name as a key for both Android and Kotlin, there is risk of conflict in KMP. `Named` would be a very good common type, but AGP SourceProvider does not extend it. I should either use `Any` or string + mapper functions, but that's even harder because the set alone is not enough to compute the key, need to know the target and compilation and whether it's the default one

Reply:
The reason why I'd like to remove this is to have as few as possible Android-specific things in KSP, as it makes maintenance much easier. We are a bit far from the end-goal though :)

Configuration names must be unique, and we already have logic in `decorateKotlinTarget(target: KotlinTarget)` that knows how to compute such name for `KotlinPlatformType.androidJvm` and `KotlinCompilation`. I think we can extract the name-computing logic from that method, and have it reused during configuration creation and then during configuration fetching. WDYT?

Reply:
@gavra0 do we really need a `String > Configuration` map then? If key is the ksp configuration name, `project.configurations` could do the job already.

Reply:
Totally correct, we can drop the map. The less code, the better :)

Reply:
Done